src/gretel_client/client.py

```diff
@@ -562,3 +564,61 @@ def get_cloud_client(prefix: str, api_key: str) -> Client:
         host=f"{prefix}.gretel.cloud",
         api_key=getpass("Enter Gretel API key: ") if prompt else api_key,
     )
+
+
+def project_from_uri(uri: str) -> Project:
+    """
+    Get a Project instance from a Gretel URI string, the
+    URI string must have the following format: 
+    gretel://[API_KEY]@HOSTNAME/PROJECT
+
+    Example::
+
+        gretel://grtu12345@api.gretel.cloud/my_project
+
+    If your API key is set as an environment variable, you may
+    omit the API key portion of the URI::
+
+        gretel://api.gretel.cloud/my_project
+    """
+    parts = urlparse(uri)
+
+    if parts.scheme != "gretel":
+        raise ValueError("URI must start with gretel://")
+
+    if not parts.netloc:
+        raise ValueError("API hostname not found")
+
+    if not parts.username:
+        username = os.getenv(DEFAULT_API_ENV_KEY)
+        if not username:
+            raise ValueError("No API key found in URI or environment")
+
+    if not parts.path or parts.path == "/":
+        raise ValueError("No project name found")
+
+    client = get_cloud_client(
+        parts.hostname.split(".")[0],
+        parts.username or username
+    )
+
+    return client.get_project(name=parts.path.strip("/"))
+
+
+@contextmanager
+def temporary_project(client: Client):
+    """Create a new project that can be used inside of a "with"
+    statement for temporary purposes. The project will be deleted
+    from Gretel Cloud when the scope is exited.
+
+    Example::
+
+        with temporary_project(client) as proj:
```

Comment:
What're your thoughts on building this as an instance method on the `Client` class? 

```
with client.temp_project() as p:
    ...
```

My thinking is it keeps the function signature a little cleaner.

Reply:
Ya that's better. I'll do that. I basically copied the example sent on Slack.

Reply:
I thought about this a little more and actually thinking keeping a module function might be more appropriate only because it provides a "side" functionality somewhat orthogonal from the other client methods since it is a context manager _and only_ a context manager. It doesn't provide any functionality directly as a callable and it feels like all the client methods that are exposed to users should maintain that contract.

I don't think it changes the amount of code to be used, either way you have to init a `Client` instance and then utilize the context manager. So it's merely to make it's functionality distinct from other instance methods.

Reply:
That's a great point. I'm good leaving it as is.